tests/test_coverage.py

```diff
@@ -14,6 +15,14 @@
 from pydantic import ValidationError
 
 
+class NdArrayInteger(NdArray):
```

Comment:
Why are these helper classes NdArrayInteger and NdArrayString needed? Maybe the validator for the type (see other comment) and an extra test in `error_cases` would be a better solution?